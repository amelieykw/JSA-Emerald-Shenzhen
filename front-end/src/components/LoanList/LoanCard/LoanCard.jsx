import React, { useContext } from 'react'
import { NavigationContext } from 'react-navigation'
import PropTypes from 'prop-types'
import { View, TouchableOpacity } from 'react-native'
import { Card, CardItem, Text, Left, Body, Right, Icon } from 'native-base'
import { connect } from 'react-redux'

import style from './LoanCardStyle'
import { setDetailId } from '../../../lib/actions'

function LoadCard({ loan, setId }) {
  const { id, type, remaining } = loan
  const navigation = useContext(NavigationContext)

  const goLoanDetails = () => {
    setId(id)
    navigation.navigate('LoanDetails', {
      transition: 'collapseExpand'
    })
  }

  const goPayNow = () => {
    navigation.navigate('PayNow', {
      transition: 'collapseExpand'
    })
  }

  return (
    <Card>
      <CardItem style={style.up}>
        <Body>
          <Text style={style.id}>{`ID - ${id}`}</Text>
          <Text style={style.type}>{type.name}</Text>
          <Text style={style.interest}>{`Current Interest rate: ${type.interest}%`}</Text>
          <Text style={style.price}>{`$ ${remaining}`}</Text>
          <Text style={style.remaining}>remaining</Text>
        </Body>
      </CardItem>
      <CardItem>
        <Left>
          <TouchableOpacity onPress={goLoanDetails}>
            <Text>DETAILS</Text>
          </TouchableOpacity>
        </Left>
        <Right>
          <TouchableOpacity style={style.button} onPress={goPayNow}>
            <View style={style.circle}>
              <Icon type="FontAwesome" name="dollar" style={style.dollar} />
            </View>
            <Text>PAY NOW</Text>
          </TouchableOpacity>
        </Right>
      </CardItem>
    </Card>
  )
}

const mapDispatchToProps = dispatch => {
  return {
    setId: id => dispatch(setDetailId(id))
  }
}

LoadCard.propTypes = {
  loan: PropTypes.objectOf(PropTypes.any).isRequired,
  setId: PropTypes.func.isRequired
}
export default connect(null, mapDispatchToProps)(LoadCard)
