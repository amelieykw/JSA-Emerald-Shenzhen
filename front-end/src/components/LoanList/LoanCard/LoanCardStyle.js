import { StyleSheet } from 'react-native'

const style = StyleSheet.create({
  loanCardUpPart: {
    backgroundColor: '#E3F2FD'
  },
  loanId: {
    fontSize: 18,
    marginTop: 10,
    color: '#7A7A7A'
  },
  loanType: {
    marginTop: 10,
    fontSize: 32,
    color: '#5A5C60'
  },
  loanCurrentRate: {
    fontSize: 15,
    color: '#000'
  },
  loanPrice: {
    fontSize: 36,
    color: '#FC815E',
    alignSelf: 'flex-end'
  },
  loanRemaining: {
    paddingRight: 5,
    fontSize: 12,
    alignSelf: 'flex-end',
    color: '#7A7A7A'
  },
  payNowButtonText: {
    color: '#000',
    margin: 0,
    paddingLeft: 20
  },
  payNowButtonImg: {
    position: 'absolute',
    left: -10
  }
})

export default style
