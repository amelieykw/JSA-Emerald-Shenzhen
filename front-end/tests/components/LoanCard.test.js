import React from 'react'
import renderer from 'react-test-renderer'
import LoanCard from '../../src/components/LoanList/LoanCard/LoanCard'
import mockData from '../../helpers/mockData_FE'

describe('<LoanHome />', () => {
  it('<LoanList > renders correctly', () => {
    const tree = renderer.create(<LoanCard loan={mockData.loans[0]} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
