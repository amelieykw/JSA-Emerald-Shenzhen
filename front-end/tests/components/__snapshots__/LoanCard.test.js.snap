// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoanCard /> <LoanCard /> should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomWidth": 0.5,
      "borderColor": "#ccc",
      "borderLeftWidth": 0.5,
      "borderRadius": 2,
      "borderRightWidth": 0.5,
      "borderTopWidth": 0.5,
      "elevation": 3,
      "flexWrap": "nowrap",
      "marginBottom": 5,
      "marginLeft": 2,
      "marginRight": 2,
      "marginTop": 5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 1.5,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        },
        Object {
          "backgroundColor": "#E3F2FD",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "stretch",
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#7A7A7A",
              "fontSize": 18,
              "marginTop": 10,
            },
          ]
        }
        uppercase={false}
      >
        ID - 1
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#5A5C60",
              "fontSize": 32,
              "marginTop": 10,
            },
          ]
        }
        uppercase={false}
      >
        mockType
      </Text>
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        Current Interest rate: undefined%
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "alignSelf": "flex-end",
              "color": "#FC815E",
              "fontSize": 36,
            },
          ]
        }
        uppercase={false}
      >
        $ 200
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "alignSelf": "flex-end",
              "color": "#7A7A7A",
              "fontSize": 12,
              "paddingRight": 5,
            },
          ]
        }
        uppercase={false}
      >
        remaining
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
              "marginLeft": 10,
            }
          }
          uppercase={false}
        >
          DETAILS
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "alignSelf": "center",
          "flex": 0.8,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "black",
              "borderRadius": 20,
              "height": 21,
              "marginRight": 8,
              "width": 21,
            }
          }
        >
          <Text />
        </View>
        <Text
          style={
            Object {
              "alignSelf": null,
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 15,
            }
          }
          uppercase={false}
        >
          PAY NOW
        </Text>
      </View>
    </View>
  </View>
</View>
`;
